File: 4-curso/2 cuatrimestre/DS/Practica 2/ejercicio_filtro/lib/FiltroContenido.dart
import 'Filtro.dart';

class FiltroContenido implements Filtro {
  late List<String> _palabrasProhibidas;

  FiltroContenido() {
    _palabrasProhibidas = [];
  }

  FiltroContenido.n2(List<String> palabras) {
    _palabrasProhibidas = palabras;
  }

  void aniadePalabras(String palabras) {
    _palabrasProhibidas.add(palabras);
  }

  @override
  String toString() {
    String aux = "Soy un filtro contenido: ";
    for (String pal in _palabrasProhibidas) {
      aux += "$pal ";
    }
    aux += "\n";

    return aux;
  }

  @override
  String ejecutar(String texto) {
    String texto_filtrado = texto;

    if (texto.isNotEmpty) {
      for (String palabra in _palabrasProhibidas) {
        texto_filtrado = texto_filtrado.replaceAll(palabra, "***");
      }
    }

    return texto_filtrado;
  }
}

File: 4-curso/2 cuatrimestre/DS/Practica 1/Ejercicios/Ejercicios 2/ejercicio2s2/src/main/java/com/p1/FiltroContenido.java
package com.p1;

import java.util.ArrayList;

public class FiltroContenido implements Filtro {
    private ArrayList<String> palabrasProhibidas;

    FiltroContenido(){
        palabrasProhibidas = new ArrayList<>();
    }

    FiltroContenido(ArrayList<String> palabras){
        palabrasProhibidas = palabras;
    }

    public void aniadePalabra(String palabra){
        palabrasProhibidas.add(palabra);
    }

    @Override
    public String ejecutar(String texto){
        String texto_filtrado = texto;

        if(!texto.isEmpty())  // Texto no vacío
            for( String palabra : palabrasProhibidas)   // Por cada palabra prohibida
                texto_filtrado = texto.replaceAll(palabra, "***");  // Busca en el texto y si está la sustituye

        return texto_filtrado;
    }
}
